letsencrypt/client/tests/display/ops_test.py

```diff
@@ -0,0 +1,183 @@
+"""Test letsencrypt.client.display.ops."""
+import sys
+import unittest
+
+import mock
+import zope.component
+
+from letsencrypt.client.display import util as display_util
+
+
+class ChooseAuthenticatorTest(unittest.TestCase):
+    """Test choose_authenticator function."""
+    def setUp(self):
+        zope.component.provideUtility(display_util.FileDisplay(sys.stdout))
+        self.mock_apache = mock.Mock()
+        self.mock_stand = mock.Mock()
+        self.mock_apache().more_info.return_value = "Apache Info"
+        self.mock_stand().more_info.return_value = "Standalone Info"
+
+        self.auths = [self.mock_apache, self.mock_stand]
+
+        self.errs = {self.mock_apache: "This is an error message."}
+
+    @classmethod
+    def _call(cls, auths, errs):
+        from letsencrypt.client.display.ops import choose_authenticator
+        return choose_authenticator(auths, errs)
+
+    @mock.patch("letsencrypt.client.display.ops.util")
+    def test_successful_choice(self, mock_util):
+        mock_util().menu.return_value = (display_util.OK, 0)
+
+        ret = self._call(self.auths, {})
+
+        self.assertEqual(ret, self.mock_apache)
+
+    @mock.patch("letsencrypt.client.display.ops.util")
+    def test_more_info(self, mock_util):
+        mock_util().menu.side_effect = [
+            (display_util.HELP, 0),
+            (display_util.HELP, 1),
+            (display_util.OK, 1),
+        ]
+
+        ret = self._call(self.auths, self.errs)
+
+        self.assertEqual(mock_util().notification.call_count, 2)
+        self.assertEqual(ret, self.mock_stand)
+
+    @mock.patch("letsencrypt.client.display.ops.util")
+    def test_no_choice(self, mock_util):
+        mock_util().menu.return_value = (display_util.CANCEL, 0)
+
+        self.assertRaises(SystemExit, self._call, self.auths, {})
+
+
+class GenHttpsNamesTest(unittest.TestCase):
+    """Test _gen_https_names."""
+    def setUp(self):
+        zope.component.provideUtility(display_util.FileDisplay(sys.stdout))
+
+    @classmethod
+    def _call(cls, domains):
+        from letsencrypt.client.display.ops import _gen_https_names
+        return _gen_https_names(domains)
+
+    def test_zero(self):
+        self.assertEqual(self._call([]), "")
+
+    def test_one(self):
+        dom = "example.com"
+        self.assertEqual(self._call([dom]), "https://%s" % dom)
+
+    def test_two(self):
+        doms = ["foo.bar.org", "bar.org"]
+        self.assertEqual(
+            self._call(doms),
+            "https://{dom[0]} and https://{dom[1]}".format(dom=doms))
+
+    def test_three(self):
+        doms = ["a.org", "b.org", "c.org"]
+        # We use an oxford comma
+        self.assertEqual(
+            self._call(doms),
+            "https://{dom[0]}, https://{dom[1]}, and https://{dom[2]}".format(
+                dom=doms))
```

Comment:
inline + less logic in tests:
```python
self.assertEqual(
    self._call(["a.org", "b.org", "c.org"]),
    "https://a.org, https://b.org, and https://c.org")
```

You can do similar in `test_four` (although I would personally drop the test case for 4 arguments)